tests/test_docs/test_bash_yaml/test_demo_docs.py

```diff
@@ -33,25 +33,26 @@
 class TestDemoDocs:
     """This class contains the tests for the bash/yaml-blocks in *.md file."""
 
+    BASH_DIR_PATH = Path(ROOT_DIR, "tests", "test_docs", "test_bash_yaml", "md_files")
+
+    def _test_blocks(self, filename: str, filter_: str):
+        """Test code blocks of a certain type determined by 'filter_' param."""
+        bash_file = Path(self.BASH_DIR_PATH, filename).read_text()
+        md_path = os.path.join(ROOT_DIR, "docs", filename.replace("bash-", ""))
+
+        bash_code_blocks = extract_code_blocks(filepath=md_path, filter=filter_)
+        for blocks in bash_code_blocks:
+            assert blocks in bash_file, "[{}]: FAILED. Code must be identical".format(
+                filename
+            )
+        logger.info("[{}]: PASSED".format(filename))
+
     def test_code_blocks_exist(self):
         """Test that all the code-blocks exist in the python file."""
-        path = Path(ROOT_DIR, "tests", "test_docs", "test_bash_yaml", "md_files")
-        logger.info(os.listdir(path))
-        for file in os.listdir(path):
+        logger.info(os.listdir(self.BASH_DIR_PATH))
+        for file in os.listdir(self.BASH_DIR_PATH):
             if not file.endswith(".md"):
                 continue
-            bash_file = read_md_file(filepath=Path(path, file))
-            md_path = os.path.join(ROOT_DIR, "docs", file.replace("bash-", ""))
-            bash_code_blocks = extract_code_blocks(filepath=md_path, filter="bash")
-            for blocks in bash_code_blocks:
-                assert (
-                    blocks in bash_file
-                ), "[{}]: FAILED. Code must be identical".format(file)
-            logger.info("[{}]: PASSED".format(file))
-
-            yaml_code_blocks = extract_code_blocks(filepath=md_path, filter="yaml")
-            for blocks in yaml_code_blocks:
-                assert (
-                    blocks in bash_file
-                ), "[{}]: FAILED. Code must be identical".format(file)
-            logger.info("[{}]: PASSED".format(file))
+            # in the future, we might add other filters like "python"
+            self._test_blocks(file, "bash")
+            self._test_blocks(file, "yaml")
```

Comment:
just a minor refactoring. I was planning to add `python` blocks, but I also think it's out of the scope of this PR.